src/diffusers/pipelines/wuerstchen/pipeline_wuerstchen.py

```diff
@@ -0,0 +1,341 @@
+
+        return text_encoder_hidden_states
+
+    def check_inputs(
+        self, predicted_image_embeddings, text_encoder_hidden_states, do_classifier_free_guidance, device
+    ):
+        if not isinstance(text_encoder_hidden_states, torch.Tensor):
+            raise TypeError(
+                f"'text_encoder_hidden_states' must be of type 'torch.Tensor', but got {type(predicted_image_embeddings)}."
+            )
+        if isinstance(predicted_image_embeddings, np.ndarray):
+            predicted_image_embeddings = torch.Tensor(predicted_image_embeddings, device=device).to(
+                dtype=text_encoder_hidden_states.dtype
+            )
+        if not isinstance(predicted_image_embeddings, torch.Tensor):
+            raise TypeError(
+                f"'predicted_image_embeddings' must be of type 'torch.Tensor' or 'np.array', but got {type(predicted_image_embeddings)}."
+            )
+
+        if do_classifier_free_guidance:
+            if predicted_image_embeddings.size(0) != text_encoder_hidden_states.size(0) // 2:
+                raise ValueError(
+                    f"'text_encoder_hidden_states' must be double the size of 'predicted_image_embeddings' in the first dimension, but {predicted_image_embeddings.size(0)} != {text_encoder_hidden_states.size(0)}."
+                )
+
+            # if predicted_image_embeddings.size(0) * 2 == text_encoder_hidden_states.size(0):
+            text_encoder_hidden_states = text_encoder_hidden_states.chunk(2)[0]
+
+            if predicted_image_embeddings.size(0) != text_encoder_hidden_states.size(0):
+                raise ValueError(
+                    f"'text_encoder_hidden_states' must be the size of 'predicted_image_embeddings' in the first dimension, but {predicted_image_embeddings.size(0)} != {text_encoder_hidden_states.size(0)}."
+                )
+
+        return predicted_image_embeddings, text_encoder_hidden_states
+
+    @torch.no_grad()
+    def __call__(
+        self,
+        image_embeds: Union[torch.FloatTensor, List[torch.FloatTensor]],
+        prompt: Union[str, List[str]] = None,
+        negative_prompt: Optional[Union[str, List[str]]] = None,
+        num_inference_steps: Union[Dict[float, int], int] = 12,
+        guidance_scale: float = 0.0,
+        num_images_per_prompt: int = 1,
+        generator: Optional[Union[torch.Generator, List[torch.Generator]]] = None,
+        latents: Optional[torch.FloatTensor] = None,
+        output_type: Optional[str] = "pil",
+        return_dict: bool = True,
+    ):
+        device = self._execution_device
+        do_classifier_free_guidance = guidance_scale > 1.0
+
+        if isinstance(image_embeds, list):
+            image_embeds = torch.cat(image_embeds, dim=0)
+        image_embeds = image_embeds.repeat_interleave(num_images_per_prompt, dim=0)
+
+        if isinstance(num_inference_steps, int):
+            num_inference_steps = {0.0: num_inference_steps}
```

Comment:
Pardon my unwisdom, but this seems like an anti-pattern to me. If it's absolutely necessary, let's add a comment.

Reply:
In my opinion this should become a standard. Diffusion models need more steps at the beginning and less steps at the end. So that's why we use 20 steps for the first 1/3 of denoising and only 10 steps for the last 2/3 of denoising as a default.

Reply:
There is almost no difference in the ouputs: Here is one generation with num_inference_steps={2 / 3: 20, 0.0: 10}:
![A_photo_of_an_astronaut_riding_a_horse_5](https://github.com/huggingface/diffusers/assets/61938694/7755f278-a889-401a-92c8-8ccdbe1710fe)
And here with num_inference_steps={0.0: 60}:
![A_photo_of_an_astronaut_riding_a_horse_6](https://github.com/huggingface/diffusers/assets/61938694/aa77d442-807f-49c8-aede-54a80dda8016)

Reply:
Actually I'm also not too happy with this pattern. We don't use this anywhere for other pipelines. The scheduling function defines quite clearly how the inference steps shall be distributed. Do we really have to provide the dictionary of `num_inference_steps`? Why can't this be handled instead with something like Karras which puts more weight on earlier timesteps or a different beta schedule?

I'd prefer to actually not allow a dictionary input to begin with

Reply:
The dict logic is quite difficult to understand for people. Can we instead opt for the IF approach of allowing custom timesteps to be passed: https://huggingface.co/docs/diffusers/api/pipelines/deepfloyd_if#optimizing-for-speed instead?

Reply:
But just because noone else is using it, does not mean that it should not be considered. If something works better than the rest, we should consider it. People can still provide a single integer and it will work as every other pipeline. We just want to save unnecessary computation and I think every scheduler can benefit from it

Reply:
Agree with you, but not the cost of breaking harmony in the library :-) 

People refer to our codebase for understanding things, and we want to be able to give them a unified and consistent experience. Plus, this deviates from the design we have for the other pipelines. So, I think it's currently better to consider a design that respects the rest of the library.

Reply:
@dome272 can't we use the IF approach where we pass custom timesteps to the pipeline?

Reply:
Aight lets do it